extra/language/nodejs.nix

```diff
@@ -0,0 +1,52 @@
+{ lib, config, pkgs, ... }:
+let
+  cfg = config.language.nodejs;
+  strOrPackage = import ../../nix/strOrPackage.nix { inherit lib pkgs; };
+in
+with lib;
+{
+  options.language.nodejs = {
+    package = mkOption {
+      type = strOrPackage;
+      default = pkgs.nodejs; # latest
+      example = literalExpression "pkgs.nodejs-18_x";
+      description = "Which nodejs package to use";
+    };
+  };
+
+  config = {
+    devshell.packages = [
+      cfg.package # nodejs itself
+
+      # TODO: npm
+
+      # Yarn
+      (pkgs.yarn.override {
+        nodejs = cfg.package;
+      })
+
+      # Pnpm
+      (pkgs.nodePackages.pnpm.override {
+        nodejs = cfg.package; # sadly doesn't suffice, but I found this:
```

Comment:
I think it could be like this:

* override nodejs pkg in `pkg.nodePackages`

```nix
let nodePackages = pkgs.nodePackages.override {
  nodejs = cfg. package
};
```

* define packages

```nix
devshell.packages = [
  cfg.package
 
  nodePackages.yarn
  nodePackages.pnpm
]
```

Reply:
I think I tried this, but don't remember.

I figured that the overlay is [much better](https://github.com/numtide/devshell/pull/233#issuecomment-1287929903) and will then cover everything dependant on nodejs (e.g. yarn toplevel)

But I don't know how to do that in a devshell extra, do you?